### PR TITLE
feat(rest): Whitelisting Fields in the REST API Response

### LINKED ISSUE #487: REST API: components: field "homepage" missing
The REST API endpoint GET components doesn't provide component's homepage.

Plus, it seems to be impossible to update it using the PATCH request.

I consider this as rather important field to clearly describe publicly available components.

### LINKED ISSUE #516: REST API: Include Vulnerability flags when requesting a single project
When requesting a single project through the `api/projects/:id` endpoint, the two flags related to vulnerabilities: 

- Enable Security Vulnerability Monitoring
- Enable Displaying Vulnerabilities

should be included in the API response.

### PR DESCRIPTION
**Pull request by smrutis1**
* White listed "homepage" field at component level.
* White listed "enableSvm" and   "enableVulnerabilitiesDIsplay" at project level.

Signed-off-by: Smruti Sahoo <smruti.sahoo@siemens.com>

closes #487 , closes #516

### CODE REVIEW

**Comment by smrutis1** [comment]
@maxhbr I have added the same in the post request. Please re-review.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/EmbeddedProject.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * 
+ * This class is used for json customization for embedded project.
+ * 
+ * @author smruti.sahoo@siemens.com
+ *
+ */
+@Relation(collectionRelation = "sw360:projects")
+public class EmbeddedProject extends Project {
```
Comment (maxhbr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/EmbeddedProject.java]: Why does a project need this `EmbeddedProject` construction, but no other resource has something similar (although embedded users or licenses are implemented?)

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java [hunk 2]
```diff
@@ -182,6 +181,97 @@ public Sw360Module() {
             @JsonProperty("id")
             abstract public String getId();
         }
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        @JsonIgnoreProperties({
```
Comment (maxhbr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java]: This is mostly duplicated code. Is there maybe a option to say, that this the list for `Project` but with additional entries? Maybe by extending the `ProjectMixin`

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/EmbeddedProject.java (hunk 1 above)
Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/EmbeddedProject.java]: With the current implementation, its little difficult to blacklist the boolean fields (like `enableSvm,` `enableVulnerabilitiesDisplay` etc.) only in the project listing through the `api/projects` endpoint and white list it, when requesting for a single project through the `api/projects/:id` endpoint . That's why I have to add a subclass for the project to achieve the same. We can add the same implementation for other resources like users or licences, may be later, when we face same type of situation. Please let me know , if you have any other idea.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java (hunk 2 above)
Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java]: Done, Thanks for the suggestion.

**Comment by nutanv1contr** [comment]
Tested the PR. Functionality is working as expected.Fields are now behaving correctly on rest request and response.

**Pull request merged**